src/history/historyActions.ts

```diff
@@ -28,9 +30,9 @@ export const getHistoryData: (
   symbol: string
 ) => ThunkResult<void, HistoryReceivedAction> = symbol => (dispatch, getState, api) => {
   const { priceFilter, dateFilter } = getState().history
-  api.getHistory(symbol, dateFilter, priceFilter).then((history: HistoryData[]) => {
-    return dispatch(historyReceivedAction(history))
-  })
+  return api
```

Comment:
again I would prefer to see these thunk actions using async/await as it is easier to read and avoids problems exactly like the one you are fixing here.